### PR TITLE
ENH Use symfony/validation logic

### PR DESCRIPTION
**Pull request by GuySartorelli**
Relies on changes in https://github.com/silverstripe/silverstripe-framework/pull/11399

## Issue
- https://github.com/silverstripe/.github/issues/193

### CODE REVIEW

File: src/Models/ExternalLink.php [hunk 1]
```diff
@@ -28,6 +28,7 @@ public function getCMSFields(): FieldList
     {
         $this->beforeUpdateCMSFields(function (FieldList $fields) {
             $field = UrlField::create('ExternalUrl', _t(__CLASS__ . '.EXTERNAL_URL_FIELD', 'External URL'));
+            $field->setAllowedProtocols(['http', 'https']);
```
Comment (ChloeSartorelli) [inline src/Models/ExternalLink.php:31]: Explicitly enforcing these protocols, even if the global config allows more protocols than this. We had decided originally this was explicitly what we wanted this link type to be used for. We can always loosen it later on if we want to.

**Review by emteknetnz — APPROVED** [review]

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Behat failure

**Comment by GuySartorelli** [comment]
Oops, I somehow missed a reference to the old `PasswordValidator`.
Should be fixed by https://github.com/silverstripe/silverstripe-behat-extension/pull/286

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Still failing after merging behat-extension PR and re-running

**Comment by GuySartorelli** [comment]
I must have been very distracted when doing that original framework PR lol. This behat failure is caused by a bug I introduced there.
Fixed by https://github.com/silverstripe/silverstripe-framework/pull/11414

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**